### ISSUE TITLE
Powershell 7 Ctrl + G from Terminal fails for paths with space

### ISSUE THREAD

**Issue by albertosantini** [microsoft/vscode#232258]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.94.2
- OS Version: Windows 11

Steps to Reproduce:

1. Open Terminal with Powershell `7.4.6`
2. mkdir "TEST TEST"
3. cd "TEST TEST"
4. cd -
5. Ctrl + G
6. Select a directory to go: `___ path ___\TEST TEST`
7. In the Terminal you see `cd & '___ path ___\TEST TEST'`

Changing directory fails due to the space in path: the command contains a weird `&`.
Other paths, withous spaces, correctly work.

Any hint?

**Comment by meganrogge** [comment]
does this work on an external terminal?

**Comment by albertosantini** [comment]
No, it doesn't work on external Terminal.

I don't see how it can work: : `cd & any_path` is executed, but it means another thing.

The point is the following: 
- when I select a path, via `Ctrl + G`, without any space, in the terminal I see the command: `cd 'C:\my\path\test'`
- when I select a path, via `Ctrl + G`, with a space, in the terminal I see the command: `cd & 'C:\my\path\test test'`

That character `&` is not correct.

**Comment by meganrogge** [comment]
Given it does not work on an external terminal, closing this as it's not an issue with VS Code.

**Comment by albertosantini** [comment]
No. :)

The problem is the shortcut Ctrl + G, key binding used in the terminal inside VS Code to change directory.

It "creates" a wrong command for the terminal.
It is a VS Code affair, not the fail of the wrong command in the external terminal.

**Comment by albertosantini** [comment]
`cd & 'C:\my\path\test test'` is a wrong command for VS Code terminal and external Terminal.
`cd'C:\my\path\test test'` is correctly executed on VS Code terminal and external Terminal.

`cd & 'C:\my\path\test test'` is created only in VS Code when the terminal has focus, typing Ctrl + G displays a combo box with paths (`select a directory to go`).

Better explanation?

**Comment by albertosantini** [comment]
I dig the code.

Here the code of the combo box, displaying the list of the paths.
`src/vs/workbench/contrib/terminalContrib/history/browser/terminalRunRecentQuickPick.ts`

Basically here it is generated the command to be executed on terminal. The path is prepared for changing folder.
```
// line 300
if (type === 'cwd') {
	text = `cd ${await instance.preparePathForShell(result.rawLabel)}`;
```

Here the util code for preparing the path for the shell
`src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts`

```
// line 319
preparePathForShell
```

In the function `preparePathForShell` we have:
```
// line 347
	if (isPowerShell && (hasSpace || originalPath.includes('\''))) {
		return `& '${originalPath.replace(/'/g, '\'\'')}'`;
	}
```

Basically those lines prepare to execute a job (`&`), not a path, a parameter for `cd` command.

The final result is the weird `&` in `cd` command.

Hope that helps.

**Comment by albertosantini** [comment]
I should debug, but I have not the setup. Sorry.


`preparePathForShell` has the following signature:

```
/**
 * Takes a path and returns the properly escaped path to send to a given shell. On Windows, this
 * included trying to prepare the path for WSL if needed.
 *
 * @param originalPath The path to be escaped and formatted.
 * @param executable The executable off the shellLaunchConfig.
 * @param title The terminal's title.
 * @param shellType The type of shell the path is being sent to.
 * @param backend The backend for the terminal.
 * @param isWindowsFrontend Whether the frontend is Windows, this is only exposed for injection via
 * tests.
 * @returns An escaped version of the path to be execuded in the terminal.
 */
```

The function is called from `terminalRunRecentQuickPick` passing only the path and executable should be `undefined`.

```
	if (!executable) {
		return originalPath;
	}
```

For some reason this condition is `false`.

**Comment by albertosantini** [comment]
@meganrogge Any chance to give a second look with the added details, please?

/cc @gjsjohnmurray
